lib/_http_client.js

```diff
@@ -636,7 +636,11 @@ ClientRequest.prototype.onSocket = function onSocket(socket) {
 function onSocketNT(req, socket) {
   if (req.aborted) {
     // If we were aborted while waiting for a socket, skip the whole thing.
-    socket.emit('free');
+    if (req.socketPath || !req.agent) {
+      socket.destroy();
+    } else {
+      socket.emit('free');
+    }
```

Comment:
I'm not even sure if it's appropriate to emit 'free' and skip destroying the socket.  `ClientRequest#abort()` always calls `socket.destroy()` when there is one.